### DISCUSSION TITLE
Using sqldb in processes

### DISCUSSION THREAD

**Discussion by KyleStiers** [nextflow-io/nf-sqldb#4] [Q&A] [answered]
I was curious if we could get some examples added to the documentation for this plugin on how best to use this within a Nextflow processes. Say you want to push Nextflow variables into a database you've defined in the config. Is it best to just do this without the plugin and just the script block? Or is there a more nextflow-y way to do this?

I think my situation is probably a common one where you want to create a channel based on a database, then push results from all of the nextflow pipeline back to a different database of results.

**Comment by pditommaso** [comment, marked as answer]
Hi, the SQL operations added by this plugin are not meant to be used *within* a process but applied to Nextlfow channels like any other operator. 

Therefore, if you want to store into a DB the data produced by a process, you need to apply it to an output channel returned by that process e.g.  


```
process yourTask {
  output: 
   tuple val(foo), val(bar)
  script:
   foo = ..
   bar = ..
   """
   some_command
   """
}

workflow {
  yourTask().sqlInsert(into: 'SAMPLE', columns: 'FOO,BAR', db: 'xyz' )
}
```

**Reply by KyleStiers** [reply]
Thanks for the prompt reply. I thought this might be the case but just wanted to clear it up. That appears to be DSL2 syntax, is that correct? Can you still do this with DSL1?

**Reply by pditommaso** [reply]
Yes, it's exactly the same concept
